src/main/java/org/gridsuite/loadflow/server/service/LoadFlowWorkerService.java

```diff
@@ -248,6 +280,211 @@ protected List<LimitViolationInfos> calculateOverloadLimitViolations(List<LimitV
+                ComponentCalculatedInfos infos = result.computeIfAbsent(Pair.of(componentValue.connectedComponentNum, componentValue.synchronousComponentNum), key -> new ComponentCalculatedInfos(0., 0., 0., 0.));
+                infos.setConsumptions(infos.getConsumptions() + componentValue.value);
+            }
+        });
+
+        // generation computation by connected/synchronous component
+        Stream.concat(network.getGeneratorStream(), network.getBatteryStream()).forEach(injection -> {
+            Terminal terminal = injection.getTerminal();
+            ComponentValue componentValue = getValueFromTerminalInComponent(terminal);
+            if (componentValue != null) {
+                ComponentCalculatedInfos infos = result.computeIfAbsent(Pair.of(componentValue.connectedComponentNum, componentValue.synchronousComponentNum), key -> new ComponentCalculatedInfos(0., 0., 0., 0.));
+                infos.setGenerations(infos.getGenerations() + abs(componentValue.value));
+            }
+        });
+
+        // exchanges computation by connected/synchronous component
+        network.getHvdcLineStream().forEach(hvdcLine -> {
+            Pair<Terminal, Terminal> terminals = getTerminalsFromIdentifiable(hvdcLine);
+            ComponentValue componentValue1 = getValueFromTerminalInComponent(terminals.getLeft());
+            ComponentValue componentValue2 = getValueFromTerminalInComponent(terminals.getRight());
+
+            if (componentValue1 != null && componentValue2 != null) {
+                ComponentCalculatedInfos infos1 = result.computeIfAbsent(Pair.of(componentValue1.connectedComponentNum, componentValue1.synchronousComponentNum), key -> new ComponentCalculatedInfos(0., 0., 0., 0.));
+                ComponentCalculatedInfos infos2 = result.computeIfAbsent(Pair.of(componentValue2.connectedComponentNum, componentValue2.synchronousComponentNum), key -> new ComponentCalculatedInfos(0., 0., 0., 0.));
+
+                if (componentValue1.connectedComponentNum != componentValue2.connectedComponentNum ||
+                    componentValue1.synchronousComponentNum != componentValue2.synchronousComponentNum) {
+                    infos1.setExchanges(infos1.getExchanges() + componentValue1.value);
+                    infos2.setExchanges(infos2.getExchanges() + componentValue2.value);
+                }
+            }
+        });
+
+        // losses computation by connected/synchronous component
+        result.forEach((key, value) ->
+            value.setLosses(value.getGenerations() - value.getConsumptions() - value.getExchanges())
+        );
+        return result;
+    }
+
+    private void fillCountryAdequacy(Map<String, CountryAdequacy> adequaciesByCountry, Country country, CountryAdequacy.ValueType valueType, double p) {
+        CountryAdequacy countryAdequacy;
+        if (adequaciesByCountry.containsKey(country.name())) {
+            countryAdequacy = adequaciesByCountry.get(country.name());
```

Comment:
the IF statement is not usefull because this is covered by computeIfEmpty() below ?

Reply:
:+1: